src/coreclr/jit/utils.cpp

```diff
@@ -4254,3 +4238,81 @@ bool CastFromDoubleOverflows(double fromValue, var_types toType)
     }
 }
 } // namespace CheckedOps
+
+template <size_t bufferSize>
+class Utf16String
+{
+private:
+    WCHAR  m_buffer[bufferSize];
+    WCHAR* m_pBuffer = nullptr;
+
+public:
+    Utf16String(const char* str)
+    {
+        int strBufferSize = MultiByteToWideChar(CP_UTF8, 0, str, -1, nullptr, 0);
+        if (strBufferSize == 0)
+        {
+            return;
```

Comment:
This is a potential bug, the contents of `m_buffer` are going to be undefined in this case. It could also be initialized, `WCHAR m_buffer[bufferSize] = {};`. Alternatively, renaming it `m_bufferUnsafe` would help assuage these concerns.

Reply:
The class encapsulates it so that the buffer pointer is only accessible via `Result()`, which returns `nullptr` in this case. I don't see any potential bugs here.

Reply:
The potential bug is someone using it and not realizing it has garbage in it. Yes, the currently implementation is correct. That isn't the issue. It is for the next person, which there are literally hundreds of issues in the runtime source base that where correct when implemented but degraded because it wasn't obvious it wasn't properly initialized. Adding the name indicates it is "unsafe" and therefore an indication it should be looked at with more scrutiny.

Reply:
Part of my defensiveness here is the countless hours I've spent fixing string issues in the runtime. The fact that we have yet another string class is an annoyance, but I get the reasoning in this case. Ideally we would be able to all agree on `SString`, but that has its own set of bugs and doesn't really work outside of the VM side.

Reply:
Sure, I will rename it.

> The fact that we have yet another string class is an annoyance, but I get the reasoning in this case.

This wasn't actually intended to be a string class.. it was more just for the RAII sake of tracking the UTF16 conversions, which got a little annoying when I had to do it for two strings.

Reply:
Yep, agree. I get why the class was made. The `SString` type should be capable of the same thing, but it is a Frankenstein monster now and does far more than it should. At some point, I'm going to fix it and have one string class in the runtime to rule them all.